source/src/drivers/cs_Storage.cpp

```diff
@@ -557,12 +557,31 @@ ret_code_t Storage::garbageCollectInternal() {
 	return fdsRetCode;
 }
 
+#if NORDIC_SDK_VERSION != 15
+/**
+ * This function can now be found as part of the example files themselves.
+ */
+static uint32_t nrf5_flash_end_addr_get()
+{
+    uint32_t const bootloader_addr = BOOTLOADER_ADDRESS;
+    uint32_t const page_sz         = NRF_FICR->CODEPAGESIZE;
+    uint32_t const code_sz         = NRF_FICR->CODESIZE;
+
+    return (bootloader_addr != 0xFFFFFFFF ?
+            bootloader_addr : (code_sz * page_sz));
+}
+#endif
+
 cs_ret_code_t Storage::eraseAllPages() {
 	LOGw("eraseAllPages");
 	if (_initialized || isErasingPages()) {
 		return ERR_NOT_AVAILABLE;
 	}
+#if NORDIC_SDK_VERSION == 15
 	uint32_t endAddr = fds_flash_end_addr();
+#else
+	uint32_t endAddr = nrf5_flash_end_addr_get();
```

Comment:
This code doesn't take `FDS_VIRTUAL_PAGES_RESERVED` into account, was that removed in the new SDK?

Reply:
See above remark. Recommendation to implement `flash_end_addr_get`in the bluenet code base and this corrects for the space taken up by fds by the product of `FDS_PHY_PAGES_RESERVED`, `FDS_PHY_PAGE_SIZE` and the individual entry size.

Reply:
This can now be patched on https://github.com/crownstone/nrF5_SDK/tree/17.1.0 itself which removes the need for an if else statement here altogether.

Reply:
Fixed on that branch and in https://github.com/crownstone/bluenet/pull/137/commits/8c9935e28732663fd9dff11015e4ad169cd3cc15